src/univers/gem.py

```diff
@@ -225,10 +220,10 @@ def equal_strictly(self, other):
         return self.version == other.version
 
     def __hash__(self):
-        return hash(self.canonical_segments)
+        return hash(tuple(list(self.canonical_segments)))
```

Comment:
Not to double call tuple + list ... but also IMHO, canonical_segments should return a tuple... see the other comment below.
```suggestion
        return hash(tuple(self.canonical_segments))
```